### PR TITLE
More async refactoring Pt 1

### PR DESCRIPTION
**Pull request by MartinJibber**

### CODE REVIEW

File: Intents Extension/Intent Handlers/FocusIntentHandler.swift [hunk 1]
```diff
@@ -37,8 +37,7 @@ class FocusIntentHandler: NSObject, INShareFocusStatusIntentHandling {
     func handle(intent: INShareFocusStatusIntent, completion: @escaping (INShareFocusStatusIntentResponse) -> Void) {
         guard let isFocused = intent.focusStatus?.isFocused, let currentUser = User.current() else { return }
 
-        #warning("Change these to enums")
-        currentUser.focusStatus = isFocused ? "focused" : "available"
+        currentUser.focusStatus = isFocused ? .focused : .available
```
Comment (MartinJibber) [inline Intents Extension/Intent Handlers/FocusIntentHandler.swift:40]: Converted focus status to an enum.

File: Shared/Extensions/Logger.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import os
+
+func logDebug(_ message: String) {
+    let logger = Logger()
+    logger.log(level: .debug, "🟡 === \(message)")
+}
```
Comment (MartinJibber) [inline Shared/Extensions/Logger.swift:15]: Convenience function for logging easily identifiable messages to the console.

File: Shared/Extensions/UI Extensions/UIWindow+Extensions.swift [hunk 3]
```diff
@@ -13,8 +13,19 @@ import UIKit
 extension UIWindow {
 
     static func topWindow() -> UIWindow? {
-        for window in UIApplication.shared.windows.reversed() {
-            if window.windowLevel == UIWindow.Level.normal && window.isKeyWindow && window.frame != CGRect.zero { return window }
+        guard let windowScene = UIApplication.shared.connectedScenes.first(where: { scene in
+            return scene is UIWindowScene
+        }) as? UIWindowScene else {
+            return nil
+        }
```
Comment (MartinJibber) [inline Shared/Extensions/UI Extensions/UIWindow+Extensions.swift:20]: Accessing the windows directly without a scene is deprecated.

File: Shared/Flows/Onboarding/CodeViewController.swift [hunk 4]
```diff
@@ -12,7 +12,6 @@ import Parse
 import TMROLocalization
 import Combine
 
-#warning("Move main actor to the base view controller.")
```
Comment (MartinJibber) [inline Shared/Flows/Onboarding/CodeViewController.swift:15]: I'm going to wait for the final version of swift 5.5 before I do any more work on this. Not worth doing at the moment.

File: Shared/Models/Parse Subclasses/Reservation.swift [hunk 5]
```diff
@@ -46,20 +46,20 @@ final class Reservation: PFObject, PFSubclassing {
         }
     }
 
-    static func getUnclaimedReservationCount(for user: User) -> Future<Int, Never> {
-        return Future { promise in
+    static func getUnclaimedReservationCount(for user: User) async -> Int {
```
Comment (MartinJibber) [inline Shared/Models/Parse Subclasses/Reservation.swift:49]: Converting to async.

File: Shared/Protocols/Objectable.swift [hunk 6]
```diff
@@ -83,27 +83,14 @@ extension Objectable where Self: PFObject {
...
-                    } else {
-                        promise(.success([]))
-                    }
-                }
-            } else {
-                promise(.success([]))
-            }
-        }
-    }
-
```
Comment (MartinJibber) [inline Shared/Protocols/Objectable.swift:101]: Now async.

File: iOS/Collection View Management /CollectionViewManager.swift [hunk 7]
```diff
@@ -246,18 +246,6 @@ class CollectionViewManager<SectionType: ManagerSectionType>: NSObject, UICollec
...
-            managerCell.collectionViewManagerWillDisplay()
-        }
-    }
-
-    func collectionView(_ collectionView: UICollectionView, didEndDisplaying cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
-        if let managerCell = cell as? CollectionViewManagerCell {
-            managerCell.collectionViewManagerDidEndDisplaying()
-        }
-    }
-
```
Comment (MartinJibber) [inline iOS/Collection View Management /CollectionViewManager.swift:260]: Will display and end display are no longer needed.

File: iOS/Flows/Channel/Manager/ChannelSupplier.swift [hunk 8]
```diff
@@ -178,12 +178,17 @@ class ChannelSupplier {
 
     // MARK: GETTERS
 
-    func waitForInitialSync() -> Future<Void, Never> {
-        return Future { promise in
-            ChannelSupplier.shared.$isSynced.mainSink { (isSynced) in
-                guard isSynced else { return }
-                promise(.success(()))
-            }.store(in: &self.cancellables)
+    func waitForInitialSync() async {
```
Comment (MartinJibber) [inline iOS/Flows/Channel/Manager/ChannelSupplier.swift:181]: The rest is moving stuff to async.

File: Jibber.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -10,6 +10,9 @@
 		2C9DC2F4715948651C293215 /* Pods_Notification_Content.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 8C8815B0EA9F1296FC57545F /* Pods_Notification_Content.framework */; };
 		6FBC3C8F68FC243D6E1A42F6 /* Pods_Notification_Service.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 6FBC4612F5436670D960DB97 /* Pods_Notification_Service.framework */; };
 		7F227B8576E160C9469E08D1 /* Pods_Intents_Extension.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 2C50C41C9E4C24DF432E0D86 /* Pods_Intents_Extension.framework */; };
+		802C448A26CC76CC00A333A4 /* Logger.swift in Sources */ = {isa = PBXBuildFile; fileRef = 802C448926CC76CC00A333A4 /* Logger.swift */; };
```
Comment (Benjamindodgson) [inline Jibber.xcodeproj/project.pbxproj:13]: What's Logger?

File: Shared/Extensions/Logger.swift (hunk 2 above)
Comment (Benjamindodgson) [reply Shared/Extensions/Logger.swift:15]: ah cool.

File: iOS/Flows/Home/HomeCollectionViewManager.swift [hunk 10]
```diff
@@ -32,25 +32,23 @@ class HomeCollectionViewManager: CollectionViewManager<HomeCollectionViewManager
...
-            }
+        async let unclaimedReservationCount = Reservation.getUnclaimedReservationCount(for: User.current()!)
+        async let initialSyncFinished: Void = ChannelSupplier.shared.waitForInitialSync()
+        async let noticesLoaded: Void = NoticeSupplier.shared.loadNotices()
+
+        let _ = await (initialSyncFinished, noticesLoaded)
+
+        self.unclaimedCount = await unclaimedReservationCount
+        let cycle = AnimationCycle(inFromPosition: .inward, outToPosition: .inward, shouldConcatenate: true, scrollToEnd: false)
+
```
Comment (Benjamindodgson) [inline iOS/Flows/Home/HomeCollectionViewManager.swift:42]: extra space

File: iOS/Flows/Home/HomeCollectionViewManager.swift [hunk 11]
```diff
@@ -32,25 +32,23 @@ class HomeCollectionViewManager: CollectionViewManager<HomeCollectionViewManager
...
+
+        self.loadSnapshot(animationCycle: cycle).mainSink { _ in
+            // Begin auto scroll
```
Comment (Benjamindodgson) [inline iOS/Flows/Home/HomeCollectionViewManager.swift]: remove comment.

**Review by Benjamindodgson — CHANGES_REQUESTED** [review]
Minor cleanup needed. Great work.

**Pull request merged**